plugins/woocommerce/src/Blocks/Utils/ProductAvailabilityUtils.php

```diff
@@ -36,13 +36,6 @@ public static function get_product_availability( $product ) {
 			$product_availability = $product->get_availability();
 		}
 
-		/**
-		 * Filters the product availability information.
-		 *
-		 * @since 9.7.0
-		 * @param array $product_availability The product availability information.
-		 * @param \WC_Product $product Product object.
-		 */
-		return apply_filters( 'woocommerce_product_availability', $product_availability, $product );
```

Comment:
It appears that we didn't have any deprecation notices before this removal, and just dropped the filter. Out of curiosity, are there any criteria in core development when we skip the deprecation step?

Reply:
The rationale is that between WC 9.7 (when it was introduced) and WC 9.8 (last released version) the filter looked like this:

```PHP
return apply_filters( 'woocommerce_product_availability', $product_availability );
```

Given that the product context wasn't passed to the filter, the filter didn't solve any real world use case and it was, in fact, useless and introduced by mistake. It was also never documented and never promoted publicly. That's why I think it should be safe to remove.

Reply:
Gotcha, thanks for explaining!